cloudbridge/cloud/providers/azure/azure_client.py

```diff
@@ -585,13 +587,52 @@ def delete_subnet(self, subnet_id):
                                              subnet_id)
         network_name = url_params.get(NETWORK_NAME)
         subnet_name = url_params.get(SUBNET_NAME)
-        result_delete = self.network_management_client \
-            .subnets.delete(
-                self.resource_group,
-                network_name,
-                subnet_name
-            )
-        result_delete.wait()
+
+        try:
+            result_delete = self.network_management_client \
+                .subnets.delete(
+                    self.resource_group,
+                    network_name,
+                    subnet_name
+                )
+            result_delete.wait()
+        except CloudError as cloud_error:
+            log.exception(cloud_error.message)
+            print(cloud_error.code)
+            if cloud_error.code != "InUseSubnetCannotBeDeleted":
+                raise cloud_error
+
+            else:
+                timeout = self._config.get("default_wait_timeout")
+                interval = self._config.get("default_wait_interval")
+
+                assert timeout >= 0
+                assert interval >= 0
+                assert timeout >= interval
+
+                end_time = time.time() + timeout
+
+                while time.time() < end_time:
```

Comment:
Is it possible to use the tenacity package here? as in this commit: https://github.com/CloudVE/cloudbridge/pull/127/commits/f1ab169b199d6ce32ea22ca83b91cdea86a2a781